internal/transport/http2_client.go

```diff
@@ -547,6 +548,10 @@ func (t *http2Client) getCallAuthData(ctx context.Context, audience string, call
 // streams.
 func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Stream, err error) {
 	ctx = peer.NewContext(ctx, t.getPeer())
+	ri := credentials.RequestInfo{
+		Method: callHdr.Method,
+	}
+	ctx = internal.NewRequestInfoContext.(func(context.Context, credentials.RequestInfo) context.Context)(ctx, ri)
```

Comment:
I didn't consider this initially, but..  This will leave the `RequestInfo` in the context for the life of the stream.  Since `RequestInfo` is in the credentials package, let's not keep it around except when calling the credentials.  If it's needed elsewhere, we might want to move `RequestInfo` somewhere else.  I would say: let's create a new context the same place where `createAudience` is called, and only pass this to the `get*AuthData` functions.

Reply:
Done. Though I had to remove the unit test that tested via NewStream. So the only test coverage is via the e2e test. That said it appears that is the case for everything in the method that calls createAudience. ¯\_(ツ)_/¯

Reply:
I prefer API-level tests to unit tests for *most* things, so that's totally fine by me.